.github/workflows/terraform-validate.yml

```diff
@@ -1,14 +1,63 @@
-name: "Terraform Validate"
+name: "Terraform Code Quality"
 
 on:
   pull_request:
   push:
 
 jobs:
-  terraform_format:
-    name: "Terraform Format"
+
+  # phase 1
+  discover_stages:
+    name: Discover
     runs-on: ubuntu-latest
+    steps:
+      # Checkout the repository to the GitHub Actions runner
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # here we create the json, we need the "id:" so we can use it in "outputs" below
+      - id: set-matrix
 
+        # find main.tf files in stages folder, trim to path, store as array, remove any empty.
+        run: echo "::set-output name=matrix::$(find stages -name main.tf | sed 's:[^/]*$::' | jq -R -s -c 'split("\n")' | jq -c 'map(select(length > 0))' )"
+        #run: echo "::set-output name=matrix::[\"stages/apps/sharedsvc/\",\"stages/apps/azure-example/\",\"stages/apps/lead/\"]"
```

Comment:
lets get rid of this. otherwise everything looks great

Reply:
Removed.